rs/boundary_node/certificate_issuance/certificate_issuer/src/acme.rs

```diff
@@ -134,28 +143,47 @@ impl Finalize for Acme {
         }
         .context("failed to generate certificate signing request")?;
 
+        // Finalize order
+        info!(domain = name, "finalizing order");
         order
             .finalize(csr.der().as_ref())
             .await
-            .context("failed to finalize order")?;
-
-        // Inject artificial delay of 5 seconds to allow certificate processing to complete
-        sleep(Duration::from_secs(5)).await;
+            .context("failed to finalize order")
+            .map_err(|e| {
+                error!(domain = name, error = %e);
+                e
+            })?;
+
+        // Poll until Valid (or timeout)
+        info!(domain = name, "polling for order valid status");
+        poll_order(&mut order, OrderStatus::Valid)
+            .await
+            .context("failed to poll order status to Valid")
+            .map_err(|e| {
+                error!(domain = name, error = %e);
+                e
+            })?;
 
         let cert_chain_pem = match order
             .certificate()
             .await
-            .context("failed to retrieve certificate")?
-        {
-            Some(cert_chain_pem) => cert_chain_pem,
+            .context("failed to retrieve certificate")
+            .map_err(|e| {
```

Comment:
Maybe cleaner would be with `.inspect_err(...)`? Or, better, inspect the error outside, where `finalize()` is called?